rotkehlchen/chain/evm/utils.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from typing import TYPE_CHECKING, Any, Callable, Optional
+
+from web3.types import BlockIdentifier
+
+from rotkehlchen.assets.asset import EvmToken
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.resolver import evm_address_to_identifier
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.fval import FVal
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import UNISWAP_PROTOCOL, VELODROME_POOL_PROTOCOL, EvmTokenKind, Price
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.types import LP_TOKEN_CONTRACT_ABI_NAMES
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+token_protocol_to_abi_name: dict[str, 'LP_TOKEN_CONTRACT_ABI_NAMES'] = {
+    UNISWAP_PROTOCOL: 'UNISWAP_V2_LP',
+    VELODROME_POOL_PROTOCOL: 'VELO_V2_LP',
+}
+
+
+def find_lp_price(
+        evm_inquirer: 'EvmNodeInquirer',
+        token: EvmToken,
+        token_price_func: Callable,
+        token_price_func_args: list[Any],
+        block_identifier: BlockIdentifier,
+        abi_name: Optional['LP_TOKEN_CONTRACT_ABI_NAMES'] = None,  # the lp token's contract abi that will be used to query the info needed for lp token price calculation  # noqa: E501
+) -> Optional[Price]:
+    """
+    Calculate the price for an LP token. That is:
+    value = (Total value of liquidity pool) / (Current supply of LP tokens)
+    We need:
+    - Price of token 0
+    - Price of token 1
+    - Pooled amount of token 0
+    - Pooled amount of token 1
+    - Total supply of pool token
+    """
+    address = token.evm_address
+    if abi_name is None:
+        abi_name = token_protocol_to_abi_name[token.protocol]
+    contract = EvmContract(
+        address=address,
+        abi=evm_inquirer.contracts.abi(abi_name),
+        deployed_block=0,
+    )
+    methods = ['token0', 'token1', 'totalSupply', 'getReserves', 'decimals']
```

Comment:
can you maybe make this a tuple?

```suggestion
    methods = ('token0', 'token1', 'totalSupply', 'getReserves', 'decimals')
```